extensions/ql-vscode/src/databases/db-config-store.ts

```diff
@@ -0,0 +1,45 @@
+import * as fs from 'fs-extra';
+import * as path from 'path';
+import { cloneDbConfig, DbConfig } from './db-config';
+
+export class DbConfigStore {
+  private readonly configPath: string;
+
+  private config: DbConfig;
+
+  public constructor(workspaceStoragePath: string) {
+    this.configPath = path.join(workspaceStoragePath, 'dbconfig.json');
+
+    this.config = emptyConfig;
+  }
+
+  public async initialize(): Promise<void> {
```

Comment:
Question: why do we have an extra initialize method for this, why not put this in the constructor? 
I wonder because by using the `initialize` method someone using the class needs to remember to call it after creating a new instance, we increase complexity. Will we ever use the class without having run what's in `initialize` before?

Reply:
It's a good point about having to remember to call `initialize()` but unfortunately constructors can't be async and we don't want to block execution while we read the config.

Reply:
Ah, you mentioned this, now I remember 🤦‍♀️ 🙈. All async methods can be replaced by methods from fs-extra I think?

Reply:
We could use sync methods but that would mean blocking things rather than doing it asynchronously. Does that make sense?